frontend/src/pages/SaveTemplate.tsx

```diff
@@ -55,8 +57,66 @@ export const SaveTemplate = () => {
         localStorage.setItem('templates', JSON.stringify([...existingTemplates, tempFile]))
         addFile(tempFile)
         }
+        
         navigate('/')
     }
+
+    const makeScreenshots = async () => {
+        const images: ImageData[] = JSON.parse(localStorage.getItem('images') || '[]') as ImageData[];
+        const shapes: CustomShape[][] = JSON.parse(localStorage.getItem('shapes') || '[]') as CustomShape[][];
+        const screenshots: string[] = [];
+    
+        const createScreenshot = (localImg: ImageData, shapeData: CustomShape[]) => {
```

Comment:
Is there a reason this function was defined inside the `makeScreenshots` function? It should likely be extracted out into a utility function to more easily facilitate unit testing.